eval/src/vespa/eval/tensor/dense/dense_tensor_value_builder.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "dense_tensor_value_builder.h"
+
+namespace vespalib::tensor {
+
+template<typename T>
+DenseTensorValueBuilder<T>::DenseTensorValueBuilder(
+        const eval::ValueType &type,
+        size_t num_mapped_in,
+        size_t subspace_size_in,
+        size_t)
+  : _type(type), 
+    _cells(subspace_size_in)
+{
+    assert(type.is_dense());
```

Comment:
I am a bit sceptical about adding too many loop-based asserts to the code we want to benchmark for fastness.